File: src/main/java/com/weeth/domain/schedule/application/mapper/EventMapper.java
package com.weeth.domain.schedule.application.mapper;

import com.weeth.domain.schedule.application.dto.ScheduleDTO;
import com.weeth.domain.schedule.domain.entity.Event;
import com.weeth.domain.user.domain.entity.User;
import org.mapstruct.*;

import static com.weeth.domain.schedule.application.dto.EventDTO.Response;

@Mapper(componentModel = MappingConstants.ComponentModel.SPRING, unmappedTargetPolicy = ReportingPolicy.IGNORE)
public interface EventMapper {

    @Mapping(target = "name", source = "event.user.name")
    @Mapping(target = "type", expression = "java(Type.EVENT)")
    Response to(Event event);

    @Mappings({
            @Mapping(target = "id", ignore = true),
            @Mapping(target = "user", source = "user")
    })
    Event from(ScheduleDTO.Save dto, User user);
}

File: src/main/kotlin/com/weeth/domain/schedule/application/mapper/EventMapper.kt
package com.weeth.domain.schedule.application.mapper

import com.weeth.domain.schedule.application.dto.request.ScheduleSaveRequest
import com.weeth.domain.schedule.application.dto.response.EventResponse
import com.weeth.domain.schedule.domain.entity.Event
import com.weeth.domain.schedule.domain.entity.enums.Type
import com.weeth.domain.user.domain.entity.User
import org.springframework.stereotype.Component

@Component
class EventMapper {
    fun toResponse(event: Event): EventResponse =
        EventResponse(
            id = event.id,
            title = event.title,
            content = event.content,
            location = event.location,
            name = event.user?.name,
            cardinal = event.cardinal,
            type = Type.EVENT,
            start = event.start,
            end = event.end,
            createdAt = event.createdAt,
            modifiedAt = event.modifiedAt,
        )

    fun toEntity(
        request: ScheduleSaveRequest,
        user: User,
    ): Event =
        Event.create(
            title = request.title,
            content = request.content,
            location = request.location,
            cardinal = request.cardinal,
            start = request.start,
            end = request.end,
            user = user,
        )
}
